File: exercism/clojure/collatz-conjecture/src/collatz_conjecture.clj
(ns collatz-conjecture)

(defn collatz [num]
  (if (<= num 0) (throw (IllegalArgumentException. "The provided number should be strictly positive"))
      (loop [n num step 0]
        (if (= 1 n)
          step
          (recur (if (= 0 (mod n 2)) (/ n 2) (+ (* n 3) 1))
                 (inc step))
          )
        )
      )
  )

File: exercism/python/collatz-conjecture/collatz_conjecture.py
def steps(number):

    if number < 1:
        raise ValueError("The given value must be a positive integer")

    step = 0
    while True:
        if number == 1:
            return step

        if number % 2 == 0:
            number = number / 2
        else:
            number = 3 * number + 1

        step += 1
